frontend/src/components/features/conversation/conversation-name.tsx

```diff
@@ -0,0 +1,122 @@
+import { EllipsisButton } from "../conversation-panel/ellipsis-button";
+import { ConversationNameContextMenu } from "./conversation-name-context-menu";
+
+export function ConversationName() {
+  const { t } = useTranslation();
+  const { conversationId } = useParams<{ conversationId: string }>();
+  const { data: conversation } = useActiveConversation();
+  const { mutate: updateConversation } = useUpdateConversation();
+
+  const [titleMode, setTitleMode] = React.useState<"view" | "edit">("view");
+  const [contextMenuOpen, setContextMenuOpen] = React.useState(false);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const handleDoubleClick = () => {
+    setTitleMode("edit");
+  };
+
+  const handleBlur = () => {
+    if (inputRef.current?.value && conversationId) {
+      const trimmed = inputRef.current.value.trim();
+      if (trimmed !== conversation?.title) {
+        updateConversation(
+          { conversationId, newTitle: trimmed },
+          {
+            onSuccess: () => {
+              displaySuccessToast(t(I18nKey.CONVERSATION$TITLE_UPDATED));
+            },
+          },
+        );
+      }
+      inputRef.current.value = trimmed;
+    } else if (inputRef.current) {
+      // reset the value if it's empty
+      inputRef.current.value = conversation?.title ?? "";
+    }
+
+    setTitleMode("view");
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      event.currentTarget.blur();
+    }
+  };
+
+  const handleInputClick = (event: React.MouseEvent<HTMLInputElement>) => {
+    if (titleMode === "edit") {
+      event.preventDefault();
+      event.stopPropagation();
+    }
+  };
+
+  const handleEllipsisClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setContextMenuOpen(!contextMenuOpen);
+  };
+
+  const handleRename = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setTitleMode("edit");
+    setContextMenuOpen(false);
+  };
+
+  React.useEffect(() => {
+    if (titleMode === "edit") {
+      inputRef.current?.focus();
+    }
+  }, [titleMode]);
+
+  if (!conversation) {
+    return null;
+  }
+
+  return (
+    <div
+      className="flex items-center gap-6 h-[22px] text-base font-normal text-left"
+      data-testid="conversation-name"
+    >
+      {titleMode === "edit" ? (
+        <input
+          ref={inputRef}
+          data-testid="conversation-name-input"
+          onClick={handleInputClick}
+          onBlur={handleBlur}
+          onKeyUp={handleKeyUp}
+          type="text"
+          defaultValue={conversation.title}
+          className="text-white leading-5 bg-transparent border-none outline-none text-base font-normal w-fit max-w-fit"
+        />
+      ) : (
+        <div
+          className="text-white leading-5 cursor-pointer w-fit max-w-fit truncate"
```

Comment:
```suggestion
          className="text-white leading-5 w-fit max-w-fit truncate"
```

Removing this since the user might mistake it for a single-click action instead of a double click one.